src/eureka/S4cal_StellarSpectra/s4cal_StellarSpec.py

```diff
@@ -0,0 +1,285 @@
+            (optspec[it0:it1], optspec[it4:it5])), axis=0)
+        std_baseline = np.ma.std(np.ma.concatenate(
+            (optspec[it0:it1], optspec[it4:it5])), axis=0)
+        # Mask outliers along wavelength axis
+        tck = spi.splrep(wave, spec_baseline, w=1/std_baseline, k=3)
+        spline = spi.splev(wave, tck)
+        mask = sigrej(spec_baseline - spline, meta.sigma_thresh)
+        igood = np.where(~mask)[0]
+        # Create XArray data arrays
+        base_flux = xrio.makeLCDA(spec_baseline[igood, np.newaxis],
+                                  wave[igood], [t0],
+                                  spec.optspec.flux_units,
+                                  spec.optspec.wave_units,
+                                  spec.optspec.time_units,
+                                  name='Median Baseline Flux')
+        base_fstd = xrio.makeLCDA(std_baseline[igood, np.newaxis],
+                                  wave[igood], [t0],
+                                  spec.optspec.flux_units,
+                                  spec.optspec.wave_units,
+                                  spec.optspec.time_units,
+                                  name='Std. Dev. of Baseline Flux')
+
+        # Median in-eclipse (or in-transit) flux
+        spec_t23 = np.ma.median(optspec[it2:it3+1], axis=0)
+        std_t23 = np.ma.std(optspec[it2:it3+1], axis=0)
+        # Mask outliers along wavelength axis
+        tck = spi.splrep(wave, spec_t23, w=1/std_t23, k=3)
+        spline = spi.splev(wave, tck)
+        mask = sigrej(spec_t23 - spline, meta.sigma_thresh)
+        igood = np.where(~mask)[0]
+        # Create XArray data arrays
+        ecl_flux = xrio.makeLCDA(spec_t23[igood, np.newaxis],
+                                 wave[igood], [t0],
+                                 spec.optspec.flux_units,
+                                 spec.optspec.wave_units,
+                                 spec.optspec.time_units,
+                                 name='Median In-Eclipse Flux')
+        ecl_fstd = xrio.makeLCDA(std_t23[igood, np.newaxis],
+                                 wave[igood], [t0],
+                                 spec.optspec.flux_units,
+                                 spec.optspec.wave_units,
+                                 spec.optspec.time_units,
+                                 name='Std. Dev. of In-Eclipse Flux')
+
+        # Create XArray dataset
+        ds = dict(base_flux=base_flux, base_fstd=base_fstd,
+                  ecl_flux=ecl_flux, ecl_fstd=ecl_fstd)
+        ds = xrio.makeDataset(ds)
+        batch.append(ds)
+
+    # Concatenate datasets
+    ds = xrio.concat(batch, dim="time")
+    filename = meta.outputdir+'S4cal_'+meta.eventlabel + "_CalStellarSpec.h5"
+    xrio.writeXR(filename, ds, verbose=True)
+
+    if meta.isplots_S4cal >= 2:
+        plot_stellarSpec(meta, ds)
+
+    return meta, spec, ds
+
+
+def plot_whitelc(optspec, time, meta, i, fig=None, ax=None):
```

Comment:
Do we want to keep these plotting functions in the same file, or move them out to a separate plots_s4cal.py file like we've done with other stages?